### DISCUSSION TITLE
Concatenating a prefix with variable doesn't generate CSS in `JIT` mode.

### DISCUSSION THREAD

**Discussion by AbdallahAbis** [tailwindlabs/tailwindcss#4180] [Help]
**What version of Tailwind CSS are you using?** 
^2.1.2
 **What build tool (or framework if it abstracts the build tool) are you using?** 
^3.0.4
 **What version of Node.js are you using?** 
12.18.4
 **What browser are you using?**
Chrome, Firefox
 **What operating system are you using?**  
MacOS

**Describe your issue** 
I'm using tailwind with gatsby v3. I enabled `JIT` by adding `mode: jit` to my `tailwind.config.js`  and I'm concatenating some variables with classNames prefixes (e.g. `px-${constants.padding}`) which should translate to `px-10`, inspecting the element does show `px-10 being applied but doesn't generate any CSS for it (just like adding a non-existing class ). doing `px-10` directly does work and generates CSS.

Tailwind.config.js:
```
module.exports = {
  mode: 'jit',
  purge: ["./src/**/*.{js,jsx,ts,tsx}", './src/**/*.js'],
 theme: {},
 variants: {
    outline: [],
  },
  plugins: [],
  corePlugins: {
    outline: false,
  }
}

**Comment by adamwathan** [comment]
Hey! This is expected, you still need to follow the rules of writing purgeable HTML:

https://tailwindcss.com/docs/optimizing-for-production#writing-purgeable-html

We don't ship any sort of client-side run time or anything so everything needs to be statically analyzable, and we just look for complete class names, we don't try to evaluate your JS or anything like that.

**Reply by AbdallahAbis** [reply]
@adamwathan Thank you for the info!
is there a way to configure Tailwind to recognize those concatenated classes?

**Reply by adamwathan** [reply]
No, it's just not technically really possible without some client-side runtime. For example if you are doing `bg-[${someValueRetrievedFromAnApi}]` it's not possible for us to know that value when compiling the CSS.

**Comment by Ojay** [comment]
Hi there, I've just discovered this exact issue/situation.

The way I've gotten around it is by dynamically adding a class, a continent name in my case, to my JSX parent element which is then referenced from a custom css file. I use the required Tailwind classes there via the `@apply` method.

Being able to use dynamic/concatenated classes would be preferable, but hey... this works just as well IMO.

**Comment by laneme** [comment]
I was quite excited about the JIT mode until I saw everything breaking when enabled. Reason: Im using the classNames lib to concat strings based on logic.
